### DISCUSSION TITLE
Reparse Paypal Description to get correct expense account

### DISCUSSION THREAD

**Discussion by greenbigfrog** [firefly-iii/firefly-iii#6424] [Firefly III]
I've imported my transactions via FinTS without any problems.
I make heavy use of Paypal though, so I'm left with quite a lot of transactions with expense account paypal.
Now, I know the very proper way would just be to also import/manage my paypal transactions, and have them be transfers into paypal asset account where again I'd then have the withdrawals into the individual expense accounts.

Now, what I'd like to do, is just (re)parse the description, and change the expense account accordingly. Most paypal transactions contain the recipient name, which a regex could easy get: `Ihr Einkauf bei (?<company>[\w .]+) `
```
00000000 . ALDI SUED, Ihr Einkauf bei ALDI SUED
000000 PP.2124.PP . VodafoneGmbH, Ihr Einkauf bei Vodafone GmbH 
 . Deutsche Post AG, Ihr Einkauf bei Deutsche Post AG 
 . GoPro Inc., Ihr Einkauf bei GoPro Inc. 
```
(PII redacted)

Is there some easy solution I'm missing?

**Comment by JC5** [comment]
You could use the API to add webhooks, but there is no regex engine in Firefly III.

**Reply by greenbigfrog** [reply]
Ok, so interact with the API so that it pulls transactions that have expense account PAYPAL and update the transaction accordingly. And then for keeping it up to date just hook it up to the webhook.
